src/app/shared/entry/doi/manage-dois/manage-dois-dialog.component.ts

```diff
@@ -90,6 +101,22 @@ export class ManageDoisDialogComponent {
     this.manageDoisDialogService.saveDoiSelection(this.entry, this.selectedOption);
   }
 
+  requestDoiAccessLink() {
+    this.manageDoisDialogService.requestDoiAccessLink(this.entry).subscribe((accessLink) => {
+      this.dockstoreDoiAccessLink = accessLink;
```

Comment:
Out of curiosity, do `workflowsService.getDOIEditLink` (called in `ngOnInit`) and `manageDoisDialogService.requestDoiAccessLink` do the same thing?

Reply:
Not quite: `workflowsService.getDOIEditLink` gets the edit link if it exists, otherwise returns a 404. It does not create edit links. `manageDoisDialogService.requestDoiAccessLink` creates an edit link if it does not exist and throws an error if it does.